spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/AppDeploymentKey.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.repository;
+
+import org.springframework.cloud.dataflow.core.ModuleDefinition;
+import org.springframework.cloud.dataflow.core.StreamDefinition;
+import org.springframework.util.Assert;
+
+/**
+ * Key used in an {@link AppDeploymentRepository}.
+ *
+ * @author Janne Valkealahti
+ */
+public class AppDeploymentKey {
```

Comment:
I initially thought to just create `Repository<String, String>` but hated idea that user would need to compile `key` manually out from `StreamDefinition` and `ModuleDefinition`. But I had similar thought when I quickly looked what a possible redis impl(to store as little as needed) might look like. As we don't need to serialize `AppDeploymentKey` back from repository, we could possibly just use its hash mapped to identifier.